File: BackEnd/Models/Transacao.cs
using BackEnd.Enums;

namespace BackEnd.Models;

public class Transacao
{
    public int Id { get; set; }
    public string Descricao { get; set; } = null!;
    public decimal Valor { get; set; }
    public TipoTransacao Tipo { get; set; }
    public int CategoriaId { get; set; }
    public Categoria Categoria { get; set; } = null!;
    public int PessoaId { get; set; }
    public Pessoa Pessoa { get; set; } = null!;
}

File: FrontEnd/src/types/Transacao.ts
export type TipoTransacao = "DESPESA" | "RECEITA";

export interface Transacao {
  id: number;
  descricao: string;
  valor: number;
  tipo: TipoTransacao;
  categoriaId: number;
  categoriaDescricao: string;
  pessoaId: number;
  pessoaNome: string;
}